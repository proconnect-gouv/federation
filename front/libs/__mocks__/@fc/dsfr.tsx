export const FranceConnectButton = jest.fn(() => <div>FranceConnectButton</div>);

export const SimpleButton = jest.fn(() => <div>SimpleButton</div>);

export const CheckboxInput = jest.fn(() => <div>CheckboxInput</div>);

export const ToggleInput = jest.fn(() => <div>ToggleInput</div>);

export const ApplicationLayout = jest.fn(() => <div>ApplicationLayout</div>);

export const LinkComponent = jest.fn(() => <div>LinkComponent</div>);

export const SearchBarComponent = jest.fn(() => <div>SearchBarComponent</div>);

export const Sizes = {
  LARGE: 'lg',
  MEDIUM: 'md',
  SMALL: 'sm',
};

export const AlertTypes = {
  ERROR: 'error',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
};

export const AlertComponent = jest.fn(() => <div>Alert</div>);

export const PaginationComponent = jest.fn(() => <div>PaginationComponent</div>);

export { ButtonTypes, IconPlacement, Priorities } from '@fc/dsfr';
