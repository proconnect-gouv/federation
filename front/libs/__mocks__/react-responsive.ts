export const useMediaQuery = jest.fn(() => true);
