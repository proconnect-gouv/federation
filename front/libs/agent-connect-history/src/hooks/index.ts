/* istanbul ignore file */

// declarative file
export * from './use-add-to-user-history.hook';
export * from './use-remove-from-user-history.hook';
export * from './use-user-history.hook';
