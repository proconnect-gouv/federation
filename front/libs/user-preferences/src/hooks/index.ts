/* istanbul ignore file */

// declarative file
export * from './use-user-preferences-api.hook';
