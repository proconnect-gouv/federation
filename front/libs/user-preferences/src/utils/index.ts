/* istanbul ignore file */

// declarative file
export * from './user-preferences-form.util';
