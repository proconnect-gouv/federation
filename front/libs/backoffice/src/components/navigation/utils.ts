import { NavigationItem } from '@fc/routing';

export const filterRouteHasOrderProperty = (
  route: NavigationItem,
): route is Required<NavigationItem> => {
  if (route.order === undefined) return false;
  return route.order >= 0;
};

export const sortNavigationRouteByOrder = (
  a: Required<NavigationItem>,
  b: Required<NavigationItem>,
): number => a.order - b.order;
