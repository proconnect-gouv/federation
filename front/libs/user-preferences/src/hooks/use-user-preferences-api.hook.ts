import axios, { AxiosError, AxiosResponse } from 'axios';
import { useCallback, useEffect, useState } from 'react';

import { useApiGet } from '@fc/common';

import {
  FormValues,
  IGetCsrfTokenResponse,
  UserPreferencesConfig,
  UserPreferencesData,
} from '../interfaces';
import { UserPreferencesService, UserPreferencesServiceInterface } from '../services';

export const validateHandlerCallback = ({ idpList }: Pick<FormValues, 'idpList'>) => {
  const isDefinedPreferences = idpList && Object.values(idpList);
  const hasError = isDefinedPreferences && !isDefinedPreferences.includes(true);

  if (!hasError) {
    return undefined;
  }

  return {
    idpList: 'error',
  };
};

export const useUserPreferencesApi = (options: UserPreferencesConfig) => {
  const userPreferences: UserPreferencesData = useApiGet({
    endpoint: options.API_ROUTE_USER_PREFERENCES,
  });

  const [submitErrors, setSubmitErrors] = useState<AxiosError | Error | undefined>(undefined);
  const [submitWithSuccess, setSubmitWithSuccess] = useState(false);
  const [formValues, setFormValues] = useState<FormValues | undefined>(undefined);

  const validateHandler = useCallback(validateHandlerCallback, []);

  const commitErrorHandler = useCallback((err: AxiosError | Error) => {
    setSubmitErrors(err);
    setSubmitWithSuccess(false);
  }, []);

  const commitSuccessHandler = useCallback(({ data }: AxiosResponse<UserPreferencesData>) => {
    const { allowFutureIdp, idpList } = UserPreferencesService.parseFormData(data);
    setFormValues({ allowFutureIdp, idpList });
    setSubmitErrors(undefined);
    setSubmitWithSuccess(true);
  }, []);

  const commit = useCallback(
    async ({
      allowFutureIdp,
      idpList,
    }: Pick<UserPreferencesServiceInterface, 'allowFutureIdp' | 'idpList'>) => {
      const {
        data: { csrfToken },
      } = await axios.get<IGetCsrfTokenResponse>(options.API_ROUTE_CSRF_TOKEN);

      const data = UserPreferencesService.encodeFormData({
        allowFutureIdp,
        csrfToken,
        idpList,
      });

      return axios
        .post<UserPreferencesData>(options.API_ROUTE_USER_PREFERENCES, data)
        .then(commitSuccessHandler)
        .catch(commitErrorHandler);
    },
    [
      commitSuccessHandler,
      commitErrorHandler,
      options.API_ROUTE_CSRF_TOKEN,
      options.API_ROUTE_USER_PREFERENCES,
    ],
  );

  useEffect(() => {
    if (userPreferences && !formValues) {
      // @NOTE
      // executed at first render only
      // when
      // if initial userPreferences has already been loaded
      // and if form values has not yet been set
      const { allowFutureIdp, idpList } = UserPreferencesService.parseFormData(userPreferences);
      setFormValues({ allowFutureIdp, idpList });
    }
  }, [userPreferences, formValues]);

  return {
    commit,
    formValues,
    submitErrors,
    submitWithSuccess,
    userPreferences,
    validateHandler,
  };
};
