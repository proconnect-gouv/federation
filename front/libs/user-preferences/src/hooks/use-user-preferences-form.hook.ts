import { useCallback, useEffect, useState } from 'react';

import { UserPreferencesData } from '../interfaces';
import { checkHasDefaultConfiguration, checkSomeIdpHasBeenChangedSinceLoading } from '../utils';

export interface useUserPreferencesFormProps {
  dirtyFields: Record<string, boolean>;
  userPreferences: UserPreferencesData;
}

export const useUserPreferencesForm = ({
  dirtyFields,
  userPreferences,
}: useUserPreferencesFormProps) => {
  const hasDefaultConfiguration = checkHasDefaultConfiguration(userPreferences);
  const someIdpHasBeenChangedSinceLoading = checkSomeIdpHasBeenChangedSinceLoading(dirtyFields);

  const [alertInfoState, setAlertInfoState] = useState({
    hasInteractedWithAlertInfo: false,
    isDisplayedAlertInfo: false,
  });

  const allowingIdPConfirmation = useCallback(() => {
    // @NOTE can not be mocked without a native re-implementation
    // istanbul ignore next
    setAlertInfoState((previous) => ({
      ...previous,
      hasInteractedWithAlertInfo: true,
      isDisplayedAlertInfo: false,
    }));
  }, [setAlertInfoState]);

  useEffect(() => {
    const shouldDisplayAlert =
      !alertInfoState.hasInteractedWithAlertInfo &&
      hasDefaultConfiguration &&
      someIdpHasBeenChangedSinceLoading;

    if (shouldDisplayAlert) {
      setAlertInfoState((previous) => ({
        ...previous,
        isDisplayedAlertInfo: true,
      }));
    }
  }, [
    someIdpHasBeenChangedSinceLoading,
    alertInfoState.hasInteractedWithAlertInfo,
    hasDefaultConfiguration,
  ]);

  return {
    alertInfoState,
    allowingIdPConfirmation,
  };
};
