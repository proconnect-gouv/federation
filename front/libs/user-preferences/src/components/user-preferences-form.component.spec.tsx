import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { useMediaQuery } from 'react-responsive';
import { mocked } from 'ts-jest/utils';

import { AlertComponent, SimpleButton, ToggleInput } from '@fc/dsfr';

import { useUserPreferencesForm } from '../hooks';
import { AllowFutureIdpSwitchLabelComponent } from './allow-future-idp-switch-label.component';
import { ServicesListComponent } from './services-list.component';
import { UserPreferencesFormComponent } from './user-preferences-form.component';

jest.mock('@fc/dsfr');
jest.mock('../hooks');
jest.mock('./services-list.component');
jest.mock('./allow-future-idp-switch-label.component');

describe('UserPreferencesFormComponent', () => {
  const userPreferencesMock = {
    allowFutureIdp: false,
    idpList: [expect.any(Object), expect.any(Object)],
  };
  const alertInfoStateMock = {
    hasInteractedWithAlertInfo: true,
    isDisplayedAlertInfo: false,
  };
  const hookResultMock = {
    alertInfoState: alertInfoStateMock,
    allowingIdPConfirmation: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot, display into a desktop viewport', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);
    mocked(useMediaQuery).mockReturnValueOnce(true);

    // when
    const { container } = render(
      <UserPreferencesFormComponent
        isDisabled
        dirtyFields={{}}
        hasValidationErrors={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, display into a mobile viewport', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);
    mocked(useMediaQuery).mockReturnValueOnce(false);

    // when
    const { container } = render(
      <UserPreferencesFormComponent
        isDisabled
        dirtyFields={{}}
        hasValidationErrors={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, display into a desktop viewport when form validation button is not disabled', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);
    mocked(useMediaQuery).mockReturnValueOnce(true);

    // when
    const { container } = render(
      <UserPreferencesFormComponent
        dirtyFields={{}}
        hasValidationErrors={false}
        isDisabled={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, display into a mobile viewport when form validation button is not disabled', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);
    mocked(useMediaQuery).mockReturnValueOnce(false);

    // when
    const { container } = render(
      <UserPreferencesFormComponent
        dirtyFields={{}}
        hasValidationErrors={false}
        isDisabled={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, display into a desktop viewport when the form has errors', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);
    mocked(useMediaQuery).mockReturnValueOnce(true);

    // when
    const { container } = render(
      <UserPreferencesFormComponent
        hasValidationErrors
        dirtyFields={{}}
        isDisabled={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, display into a mobile viewport when the form has errors', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);
    mocked(useMediaQuery).mockReturnValueOnce(false);

    // when
    const { container } = render(
      <UserPreferencesFormComponent
        hasValidationErrors
        dirtyFields={{}}
        isDisabled={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(container).toMatchSnapshot();
  });

  it('should call ServicesListComponent with params', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);

    // when
    render(
      <UserPreferencesFormComponent
        isDisabled
        dirtyFields={{}}
        hasValidationErrors={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(ServicesListComponent).toHaveBeenCalledTimes(1);
    expect(ServicesListComponent).toHaveBeenCalledWith(
      { identityProviders: userPreferencesMock.idpList },
      {},
    );
  });

  it('should call ToggleInput with params', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);

    // when
    render(
      <UserPreferencesFormComponent
        isDisabled
        dirtyFields={{}}
        hasValidationErrors={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(ToggleInput).toHaveBeenCalledTimes(1);
    expect(ToggleInput).toHaveBeenCalledWith(
      expect.objectContaining({
        initialValue: false,
        label: expect.any(Function),
        legend: { checked: 'Autorisé', unchecked: 'Bloqué' },
        name: 'allowFutureIdp',
      }),
      {},
    );
  });

  it('should render AllowFutureIdpSwitchLabelComponent with params, when labelCallback is called', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);
    const useCallbackMock = jest.spyOn(React, 'useCallback');

    // when
    const { container } = render(
      <UserPreferencesFormComponent
        isDisabled
        dirtyFields={{}}
        hasValidationErrors={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    act(() => {
      const callback = useCallbackMock.mock.calls[0][0];
      ReactDOM.render(<div>{callback(false)}</div>, container);
    });

    // then
    expect(AllowFutureIdpSwitchLabelComponent).toHaveBeenCalledTimes(1);
    expect(AllowFutureIdpSwitchLabelComponent).toHaveBeenCalledWith({ checked: false }, {});
  });

  it('should call AlertComponent with params when the form has errors', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);
    const expectedProps = {
      children: expect.any(Array),
      type: 'error',
    };

    // when
    render(
      <UserPreferencesFormComponent
        hasValidationErrors
        dirtyFields={{}}
        isDisabled={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(AlertComponent).toHaveBeenCalledTimes(1);
    expect(AlertComponent).toHaveBeenCalledWith(expectedProps, {});
  });

  it('should call useUserPreferencesForm with dirtyFields and userPreference when allowingIdPConfirmation is called', () => {
    // given
    mocked(AlertComponent).mockImplementationOnce(({ children }) => <div>{children}</div>);
    mocked(useUserPreferencesForm).mockReturnValue({
      alertInfoState: {
        hasInteractedWithAlertInfo: false,
        isDisplayedAlertInfo: true,
      },
      allowingIdPConfirmation: jest.fn(),
    });
    const expectedArgs = {
      dirtyFields: {},
      userPreferences: userPreferencesMock,
    };

    // when
    const { getByTestId } = render(
      <UserPreferencesFormComponent
        dirtyFields={{}}
        hasValidationErrors={false}
        isDisabled={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );
    const button = getByTestId('UserPreferenceFormComponent-button-info');
    fireEvent.click(button);

    // then
    expect(useUserPreferencesForm).toHaveBeenCalledTimes(1);
    expect(useUserPreferencesForm).toHaveBeenCalledWith(expectedArgs);
  });

  it('should not call AlertComponent if there are no errors', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);

    // when
    render(
      <UserPreferencesFormComponent
        dirtyFields={{}}
        hasValidationErrors={false}
        isDisabled={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(AlertComponent).toHaveBeenCalledTimes(0);
  });

  it('should call SimpleButton with params, can not submit', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);

    // when
    render(
      <UserPreferencesFormComponent
        isDisabled
        dirtyFields={{}}
        hasValidationErrors={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(SimpleButton).toHaveBeenCalledTimes(1);
    expect(SimpleButton).toHaveBeenCalledWith(
      expect.objectContaining({
        disabled: true,
        label: 'Enregistrer mes réglages',
        type: 'submit',
      }),
      {},
    );
  });

  it('should call SimpleButton with params, can submit', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);

    // when
    render(
      <UserPreferencesFormComponent
        dirtyFields={{}}
        hasValidationErrors={false}
        isDisabled={false}
        showNotification={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );

    // then
    expect(SimpleButton).toHaveBeenCalledTimes(1);
    expect(SimpleButton).toHaveBeenCalledWith(
      expect.objectContaining({
        disabled: false,
        label: 'Enregistrer mes réglages',
        type: 'submit',
      }),
      {},
    );
  });

  it('should show a notification when the form has been submitted', () => {
    // given
    mocked(useUserPreferencesForm).mockReturnValueOnce(hookResultMock);

    // when
    const { getByText } = render(
      <UserPreferencesFormComponent
        isDisabled
        showNotification
        dirtyFields={{}}
        hasValidationErrors={false}
        userPreferences={userPreferencesMock}
        onSubmit={jest.fn()}
      />,
    );
    const element = getByText(
      'Une notification récapitulant les modifications va vous être envoyée',
    );

    // then
    expect(element).toBeInTheDocument();
  });
});
