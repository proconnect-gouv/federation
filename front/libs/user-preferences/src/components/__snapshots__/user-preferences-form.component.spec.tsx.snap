// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPreferencesFormComponent should match the snapshot, display into a desktop viewport 1`] = `
<div>
  <form
    data-testid="user-preferences-form"
    id="UserPreferencesFormComponent"
  >
    <h2
      class="title fr-h3 fr-mt-5w"
    >
      <b>
        Mes réglages :
      </b>
    </h2>
    <p
      class="fr-mt-2w"
    >
      Attention : 
      <strong>
        Vous devez avoir au moins un compte autorisé
      </strong>
       pour continuer à utiliser FranceConnect. Nous vous conseillons de ne bloquer que les comptes que vous n’utilisez pas.
    </p>
    <div>
      ServicesListComponent
    </div>
    <p
      class="fr-mt-5w"
    >
      Pour vous offrir toujours plus de choix, il est possible que FranceConnect mette à votre disposition de nouveaux moyens d’identification dans le futur. En cochant cette case, ils ne pourront pas être utilisés pour accéder aux services proposant FranceConnect.
    </p>
    <p
      class="fr-mt-2w"
    >
      Vous pourrez les autoriser depuis cette page.
    </p>
    <div>
      CheckboxInput
    </div>
    <div
      class="text-center fr-mt-11w"
    >
      <div>
        SimpleButton
      </div>
    </div>
  </form>
</div>
`;

exports[`UserPreferencesFormComponent should match the snapshot, display into a desktop viewport when form validation button is not disabled 1`] = `
<div>
  <form
    data-testid="user-preferences-form"
    id="UserPreferencesFormComponent"
  >
    <h2
      class="title fr-h3 fr-mt-5w"
    >
      <b>
        Mes réglages :
      </b>
    </h2>
    <p
      class="fr-mt-2w"
    >
      Attention : 
      <strong>
        Vous devez avoir au moins un compte autorisé
      </strong>
       pour continuer à utiliser FranceConnect. Nous vous conseillons de ne bloquer que les comptes que vous n’utilisez pas.
    </p>
    <div>
      ServicesListComponent
    </div>
    <p
      class="fr-mt-5w"
    >
      Pour vous offrir toujours plus de choix, il est possible que FranceConnect mette à votre disposition de nouveaux moyens d’identification dans le futur. En cochant cette case, ils ne pourront pas être utilisés pour accéder aux services proposant FranceConnect.
    </p>
    <p
      class="fr-mt-2w"
    >
      Vous pourrez les autoriser depuis cette page.
    </p>
    <div>
      CheckboxInput
    </div>
    <div
      class="text-center fr-mt-11w"
    >
      <div>
        SimpleButton
      </div>
    </div>
  </form>
</div>
`;

exports[`UserPreferencesFormComponent should match the snapshot, display into a desktop viewport when the form has errors 1`] = `
<div>
  <form
    data-testid="user-preferences-form"
    id="UserPreferencesFormComponent"
  >
    <h2
      class="title fr-h3 fr-mt-5w"
    >
      <b>
        Mes réglages :
      </b>
    </h2>
    <p
      class="fr-mt-2w"
    >
      Attention : 
      <strong>
        Vous devez avoir au moins un compte autorisé
      </strong>
       pour continuer à utiliser FranceConnect. Nous vous conseillons de ne bloquer que les comptes que vous n’utilisez pas.
    </p>
    <div>
      ServicesListComponent
    </div>
    <div>
      AlertMessageComponent
    </div>
    <p
      class="fr-mt-5w"
    >
      Pour vous offrir toujours plus de choix, il est possible que FranceConnect mette à votre disposition de nouveaux moyens d’identification dans le futur. En cochant cette case, ils ne pourront pas être utilisés pour accéder aux services proposant FranceConnect.
    </p>
    <p
      class="fr-mt-2w"
    >
      Vous pourrez les autoriser depuis cette page.
    </p>
    <div>
      CheckboxInput
    </div>
    <div
      class="text-center fr-mt-11w"
    >
      <div>
        SimpleButton
      </div>
    </div>
  </form>
</div>
`;

exports[`UserPreferencesFormComponent should match the snapshot, display into a mobile viewport 1`] = `
<div>
  <form
    data-testid="user-preferences-form"
    id="UserPreferencesFormComponent"
  >
    <h2
      class="title fr-h3 fr-mt-5w"
    >
      <b>
        Mes réglages :
      </b>
    </h2>
    <p
      class="fr-mt-2w"
    >
      Attention : 
      <strong>
        Vous devez avoir au moins un compte autorisé
      </strong>
       pour continuer à utiliser FranceConnect. Nous vous conseillons de ne bloquer que les comptes que vous n’utilisez pas.
    </p>
    <div>
      ServicesListComponent
    </div>
    <p
      class="fr-mt-5w"
    >
      Pour vous offrir toujours plus de choix, il est possible que FranceConnect mette à votre disposition de nouveaux moyens d’identification dans le futur. En cochant cette case, ils ne pourront pas être utilisés pour accéder aux services proposant FranceConnect.
    </p>
    <p
      class="fr-mt-2w"
    >
      Vous pourrez les autoriser depuis cette page.
    </p>
    <div>
      CheckboxInput
    </div>
    <div
      class="text-center fr-mt-5w"
    >
      <div>
        SimpleButton
      </div>
    </div>
  </form>
</div>
`;

exports[`UserPreferencesFormComponent should match the snapshot, display into a mobile viewport when form validation button is not disabled 1`] = `
<div>
  <form
    data-testid="user-preferences-form"
    id="UserPreferencesFormComponent"
  >
    <h2
      class="title fr-h3 fr-mt-5w"
    >
      <b>
        Mes réglages :
      </b>
    </h2>
    <p
      class="fr-mt-2w"
    >
      Attention : 
      <strong>
        Vous devez avoir au moins un compte autorisé
      </strong>
       pour continuer à utiliser FranceConnect. Nous vous conseillons de ne bloquer que les comptes que vous n’utilisez pas.
    </p>
    <div>
      ServicesListComponent
    </div>
    <p
      class="fr-mt-5w"
    >
      Pour vous offrir toujours plus de choix, il est possible que FranceConnect mette à votre disposition de nouveaux moyens d’identification dans le futur. En cochant cette case, ils ne pourront pas être utilisés pour accéder aux services proposant FranceConnect.
    </p>
    <p
      class="fr-mt-2w"
    >
      Vous pourrez les autoriser depuis cette page.
    </p>
    <div>
      CheckboxInput
    </div>
    <div
      class="text-center fr-mt-5w"
    >
      <div>
        SimpleButton
      </div>
    </div>
  </form>
</div>
`;

exports[`UserPreferencesFormComponent should match the snapshot, display into a mobile viewport when the form has errors 1`] = `
<div>
  <form
    data-testid="user-preferences-form"
    id="UserPreferencesFormComponent"
  >
    <h2
      class="title fr-h3 fr-mt-5w"
    >
      <b>
        Mes réglages :
      </b>
    </h2>
    <p
      class="fr-mt-2w"
    >
      Attention : 
      <strong>
        Vous devez avoir au moins un compte autorisé
      </strong>
       pour continuer à utiliser FranceConnect. Nous vous conseillons de ne bloquer que les comptes que vous n’utilisez pas.
    </p>
    <div>
      ServicesListComponent
    </div>
    <div>
      AlertMessageComponent
    </div>
    <p
      class="fr-mt-5w"
    >
      Pour vous offrir toujours plus de choix, il est possible que FranceConnect mette à votre disposition de nouveaux moyens d’identification dans le futur. En cochant cette case, ils ne pourront pas être utilisés pour accéder aux services proposant FranceConnect.
    </p>
    <p
      class="fr-mt-2w"
    >
      Vous pourrez les autoriser depuis cette page.
    </p>
    <div>
      CheckboxInput
    </div>
    <div
      class="text-center fr-mt-5w"
    >
      <div>
        SimpleButton
      </div>
    </div>
  </form>
</div>
`;
